lib/portfinder.js

```diff
@@ -76,40 +76,52 @@ exports.getPort = function (options, callback) {
     options = {};
   }
 
-  options.host = options.host || '::1';
-  var hosts = [options.host, '0.0.0.0', '127.0.0.1'];
+  var hosts = ['::1', '127.0.0.1', '0.0.0.0'];
+  if (options.host) {
+    hosts.push(options.host);
+  }
 
   // remove duplicate host arg if already in hosts array
-  for (var i = 1; i < hosts.length; i++) {
-    if (options.host === hosts[i]) {
-      hosts.shift();
-      break;
-    }
-  }
+  var uniqueHosts = hosts.filter(function(host, index) {
+    return hosts.indexOf(host) == index;
+  });
 
-  // create array of eventual found ports for each host
-  var testPortFunctions = [];
-  for (var j = 0; j < hosts.length; j++) {
-    (function() {
-      var jj = j;
-      var testPortFn = function(callback) {
-        internals.testPort({ host: hosts[jj], port: options.port }, callback);
-      };
-
-      testPortFunctions.push(testPortFn);
-    }());
+  function findCommonOpenPort(portToTest) {
+    var ports = [];
+    var error;
+    async.everyLimit(uniqueHosts, 1, function(host, next) {
+      internals.testPort({"host": host, "port": portToTest}, function(err, port) {
+        if (err) {
+          error = err;
+          return next(false); 
+        } else {
+          ports.push(port);
+		  next(port);
+        }
+      });
+    }, function(result) {
+      if (!result) {
+        return callback(error);
+      } 
+
+      var uniquePorts = ports.filter(function(port, index) {
+        return ports.indexOf(port) == index;
+      });
+      if (uniquePorts.length > 1) {
+        // find highes port
+        uniquePorts.sort(function(a, b) {
+          return a - b;
+        });
+        // test all of the hosts against highest port
+        findCommonOpenPort(uniquePorts.pop());
+      } else {
+        callback(null, ports[0]);
+      }
+      
+    });
   }
 
-  async.series(testPortFunctions, function(err, ports) {
-    if (err) { return callback(err); }
-
-    var port = ports.pop();
-    for (var i = 0; i < ports.length; i++) {
-      if (port !== ports[i]) { return exports.getPort({ host: options.hosts, port: ports[i] }, callback); }
-    }
-
-    return callback(null, port);
-  });
+  findCommonOpenPort(options.port || exports.basePort);
```

Comment:
the internals.testPort function already starts at the port given, the exports.nextPort(i think is the name) simply adds +=1 from there, so we can remove this line (if you found differently, let me know, going off memory)